### PR TITLE
416 replace sys argv with argparse

### LINKED ISSUE #416: Replace create_plugin_registries 'sys.argv' calls with 'argparse' library
# Requested Update

### Description
Currently, create_plugin_registries.py uses sys.argv to parse arguments for this program. We should avoid using this in all of GeoIPS, as it follow the conventions that will be upheld by the new CLI. Argparse is more dynamic, and can provide useful feedback to the user for how to use a certain program.

### Background and Motivation
Lots of GeoIPS (and the eventual CLI) uses argparse, and we should adhere to this convention for all GeoIPS code.

### Alternative Solutions
Leave the code as is and deal with it. 
  
### Environment
*  [ IF APPROPRIATE: Corresponding repository versions ]

### Code to demonstrate issue
geoips/geoips/create_plugin_registries.py

### Checklist for Completion
* [ ]  Change sys.argv calls to argparse calls.

### PR DESCRIPTION
**Pull request by evrose54**
Reviewer Checklist

* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [x] NO REQUIRED ***unit tests*** (explain why not required)
* [x] NO REQUIRED ***integration tests*** (explain why not required)
* [x] Required ***documentation*** added for new/modified functionality
* [x] Required ***release notes*** added for new/modified functionality
* [x] NO REQUIRED ***updates to other repos*** (explain why not required)

https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes #416 

# Summary
Currently create_plugin_registries.py uses sys.argv calls to generate its arguments
rather than argparse. This doesn't follow GeoIPS conventions nor that of the CLI that
will soon be updated. We should replace these calls with argparse architecture, so that
this code follows current conventions, as does other GeoIPS code.

### CODE REVIEW

File: docs/source/releases/v1_11_7a0.rst [hunk 1]
```diff
@@ -32,6 +32,7 @@ Version 1.11.7a0 (2023-10-26)
 * Added unit test configuration file
```
Comment (mindyls) [inline docs/source/releases/v1_11_7a0.rst:32]: Opening conversation to remind us to NOT merge this until v1.12.0-release has been merged to main, and this destination branch is automatically updated to point to main.

**Review by jsolbrig — APPROVED** [review]

File: geoips/create_plugin_registries.py [hunk 2]
```diff
@@ -848,9 +849,30 @@ def main():
...
+        description=description,
+    )
+    argparser.add_argument(
+        "-s",
+        "--save_type",
+        type=str.lower,
+        default="json",
+        choices=["json", "yaml"],
+        help="Format to write registries to. This will also be the file extension.",
+    )
```
Comment (mindyls) [inline geoips/create_plugin_registries.py:872]: There is an incoming change that allows passing a package name into create_plugin_registries - we will update this to allow a "-p" option when 1.12.1 is released.

**Review by mindyls — APPROVED** [review]
Looks good to me.  Note my comment about incoming package argument (allowing calling create_plugin_registires with only a single package).  Should be an easy update.

File: geoips/create_plugin_registries.py (hunk 2 above)
Comment (jsolbrig) [reply geoips/create_plugin_registries.py:872]: Sounds good. I think we can handle this when 1.12.1 comes through. I'm going to resolve and merge.

**Review by jsolbrig — APPROVED** [review]

**Pull request merged**